application/src/main/java/org/togetherjava/tjbot/commands/SlashCommand.java

```diff
@@ -84,16 +84,16 @@
      * <p>
      * <p>
      * This method may be called multiple times, implementations must not create new data each time
-     * but instead configure it once beforehand. The command system will automatically call this
-     * method to register the command to Discord.
+     * but instead configure it once beforehand. The core system will automatically call this method
+     * to register the command to Discord.
      *
      * @return the command data of this command
      */
     @NotNull
     CommandData getData();
 
     /**
-     * Triggered by the command system after system startup is complete. This can be used for
+     * Triggered by the core system after system startup is complete. This can be used for
      * initialisation actions that cannot occur during construction.
```

Comment:
Agreed. So you are suggestion to remove this `onReady` method and instead let current users (probably only the help system) use the alternatives that are now available?

Reply:
no, the onready still has use / purpose

for example routine support could use it to log:
'INFO routine system loaded: 2 routines rolled back from invalid termination, 23 routines restarted, 5 old routines deleted'

we just need to remove the statement in the javadoc saying: 'use this to start things the bot doesnt support yet'

Reply:
Routines are not slash commands though. Routines will use `Routine` (#346 ). So it should then rather be moved into that interface then.

But I get your point and agree, the Javadoc has to be adjusted.

Reply:
I removed `onReady` now. The free-system now implements `EventReceiver` (instead of `EventListener`), which also gets a `ReadyEvent` handed over. And now that the `BotCore` adds it to `jda.addEventListener(...)`, it also gets triggered early enough. Seems more elegant that way.

Reply:
it looks like that will work.

technically what should happen is the JDA should be passed in on construction and the onReady code called in the constructor and then there is no need for generic events or the onready event (it was just being used as a hook to access the JDA without having to rewrite all of the code in slashadapter because commands didnt have access to JDA, Feature does.)

because the slashadapter and the messageadapter covers all functionality.

Reply:
but then again ...... maybe it should be left in `onReady`?
the Features construction is not threaded so heavy code there will block the bot's startup
and whether onReady is called by the JDA or by our bots router, it is threaded there.

Reply:
i feel like once this is merged, the free command can be simplified and made nicer but i leave that up to u for another PR. it would be too much for this PR to adjust /free beyond "just making it work", i think.